daemon/vpn/nordlynx/kernel_space.go

```diff
@@ -145,10 +147,19 @@ func (k *KernelSpace) stop() error {
 
 	k.active = false
 	k.tun = nil
-	k.state = vpn.ExitedState
+	k.setState(vpn.ExitedState)
 	return nil
 }
 
+func (k *KernelSpace) StateChanged() <-chan vpn.State {
+	return k.stateChanged
+}
+
+func (k *KernelSpace) setState(newState vpn.State) {
+	k.state = newState
```

Comment:
What if same state is being set, so, no change happened why bother informing about that someone